File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = []
        def helper(num,ind):
            if ind == len(nums):
                result.append(num[:])
                return 
            helper(num,ind+1)
            num.append(nums[ind])
            helper(num,ind+1)
            if num:
                num.pop()
        helper([],0)
        return result



        

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n=nums.size();
        int subset=pow(2,n);
        vector<vector<int>>ans;
        for(int i=0;i<=subset-1;i++)
        {
            vector<int>ds;
            for(int j=0;j<=n-1;j++)
            {
                if(i&(1<<j))
                {
                  ds.push_back(nums[j]);  
                }
            }
        ans.push_back(ds);
        }
        return ans;
    }
};